futures-util/src/future/future/mod.rs

```diff
@@ -112,6 +111,36 @@ mod shared;
 #[cfg(any(feature = "std", all(feature = "alloc", feature = "spin")))]
 pub use self::shared::{Shared, WeakShared};
 
+/// Pins a value on the stack.
+///
+/// **Note:** Since Rust 1.68, this macro is soft-deprecated in favor of
+/// [`pin!`](https://doc.rust-lang.org/std/pin/macro.pin.html) macro
+/// in the standard library.
+///
+/// # Example
+///
+/// ```rust
+/// # use futures_util::pin_mut;
+/// # use core::pin::Pin;
+/// # struct Foo {}
+/// let foo = Foo { /* ... */ };
+/// pin_mut!(foo);
+/// let _: Pin<&mut Foo> = foo;
+/// ```
+#[macro_export]
+macro_rules! pin_mut {
+    ($($x:ident),* $(,)?) => { $(
+        // Move the value to ensure that it is owned
+        let mut $x = $x;
+        // Shadow the original binding so that it can't be directly accessed
+        // ever again.
+        #[allow(unused_mut)]
+        let mut $x = unsafe {
+            ::core::pin::Pin::new_unchecked(&mut $x)
```

Comment:
This should refer `$crate::__private::Pin` instead of `::core::pin::Pin`, [as pin-utils does](https://github.com/rust-lang/pin-utils/blob/c635c0ca87cd9f9173152f367ef1d84df45943ab/src/stack_pin.rs#L27).

Reply:
Any rationale here?

I'm afraid that, as `futures_util::__private` is guarded by `async-await` flag, it would create an extra flag gate to use this name.

Reply:
> Any rationale here?


See https://github.com/rust-lang/pin-utils/pull/26#discussion_r344491597.



> I'm afraid that, as `futures_util::__private` is guarded by `async-await` flag, it would create an extra flag gate to use this name.

There is no problem with removing the cfg from `__private` module.
That cfg just indicates who the current user is, nothing special.